invokeai/app/services/shared/invocation_context.py

```diff
@@ -324,9 +328,13 @@ def load(self, key: str, submodel_type: Optional[SubModelType] = None) -> Loaded
         # The model manager emits events as it loads the model. It needs the context data to build
         # the event payloads.
 
-        return self._services.model_manager.load_model_by_key(
-            key=key, submodel_type=submodel_type, context_data=self._data
-        )
+        if isinstance(identifier, str):
+            model = self._services.model_manager.store.get_model(identifier)
+            return self._services.model_manager.load.load_model(model, submodel_type, self._data)
+        else:
+            _submodel_type = submodel_type or identifier.submodel_type
+            model = self._services.model_manager.store.get_model(identifier.key)
```

Comment:
Is this were we were considering interpreting the key as either a hash or an ID? Maybe something like this?

```
  model: AnyModelConfig
  key = identifier if isinstance(identifier, str) else identifier.key
  submodel_type = submodel_type\
      if not (isinstance(identifier, ModelField) and identifier.submodel_type)\
      else identifier.submodel_type
  try:
    model = self._services.model_manager.store.get_model(key)
  except UnknownModelException:
    model = self._services.model_manager.store. get_model_by_hash(key)
  return self._services.model_manager.load.load_model(model, submodel_type, self._data)
```

Reply:
No, I realized we don't need to load models by hash. The client will handle reconciling metadata and retrieving the key for a model. Then the backend only loads by key.

Reply:
I suppose not 🤔 I guess not a lot of people will interface with the API outside of the UI